src/dist/http.rs

```diff
@@ -670,10 +670,21 @@ mod server {
             macro_rules! check_server_auth_or_401 {
                 ($request:ident) => {{
                     match bearer_http_auth($request).and_then(&*check_server_auth) {
-                        Some(server_id) if server_id.addr().ip() == $request.remote_addr().ip() => {
-                            server_id
-                        }
-                        Some(_) => return make_401("invalid_bearer_token_mismatched_address"),
+                        Some(server_id) => {
+                            let origin_ip = if let Some(header_val) = $request.header("X-Real-IP") {
+                                trace!("X-Real-IP: {:?}", header_val);
+                                header_val.parse().unwrap()
```

Comment:
I think it'd be more robust to handle the error case here rather than blindly calling `unwrap()`.  I'm not sure whether you can `header_val.parse()?` or something more verbose is required.